### PR TITLE
Refine error message in pybind folder

### PR DESCRIPTION
**Pull request by zhiqiu**
### PR types

Others
### PR changes

Others
### Describe

Refine error message in `pybind` folder

### CODE REVIEW

File: paddle/fluid/pybind/data_set_py.cc [hunk 1]
```diff
@@ -80,14 +86,19 @@ class IterableDatasetWrapper {
...
-                      "Device number does not match reader number");
+                      platform::errors::InvalidArgument(
+                          "Device number does not match reader number"));
     for (size_t i = 0; i < places_.size(); ++i) {
       data_feeds_[i]->AssignFeedVar(*scopes_[i]);
       data_feeds_[i]->SetPlace(platform::CPUPlace());
-      PADDLE_ENFORCE_EQ(data_feeds_[i]->Start(), true, "Reader start failed");
+      PADDLE_ENFORCE_EQ(
+          data_feeds_[i]->Start(), true,
+          platform::errors::Unavailable("Failed to start the reader."));
```
Comment (chenwhql) [inline paddle/fluid/pybind/data_set_py.cc]: maybe we can tell users which card‘s reader start failed？

File: paddle/fluid/pybind/data_set_py.cc [hunk 2]
```diff
@@ -96,7 +107,8 @@ class IterableDatasetWrapper {
   }
 
   std::vector<std::unordered_map<std::string, framework::LoDTensor>> Next() {
-    PADDLE_ENFORCE_EQ(is_started_, true, "Reader must be started");
+    PADDLE_ENFORCE_EQ(is_started_, true, platform::errors::PreconditionNotMet(
+                                             "Reader must be started"));
```
Comment (chenwhql) [inline paddle/fluid/pybind/data_set_py.cc]: Reader must be started when get next batch data?

File: paddle/fluid/pybind/data_set_py.cc [hunk 3]
```diff
@@ -154,7 +166,9 @@ class IterableDatasetWrapper {
  private:
   bool IsValidLoDTensor(const framework::LoDTensor &tensor) const {
     auto &lod = tensor.lod();
-    PADDLE_ENFORCE_LE(lod.size(), 1, "lod level must be not larger than 1");
+    PADDLE_ENFORCE_LE(lod.size(), 1,
+                      platform::errors::InvalidArgument(
+                          "lod level must be not larger than 1"));
```
Comment (chenwhql) [inline paddle/fluid/pybind/data_set_py.cc]: maybe this `lod` can alse use uppercase, such as `Lod` or `LoD`?

File: paddle/fluid/pybind/pybind.cc [hunk 4]
```diff
@@ -175,7 +175,8 @@ static T PyObjectCast(PyObject *obj) {
   try {
     return py::cast<T>(py::handle(obj));
   } catch (py::cast_error &) {
-    PADDLE_THROW("Python object is not type of %s", typeid(T).name());
+    PADDLE_THROW(platform::errors::InvalidArgument(
+        "Python object is not type of %s", typeid(T).name()));
```
Comment (chenwhql) [inline paddle/fluid/pybind/pybind.cc]: can we get actual type of PyObject?

File: paddle/fluid/pybind/pybind.cc [hunk 5]
```diff
@@ -659,20 +671,22 @@ PYBIND11_MODULE(core_noavx, m) {
...
+            LoD new_lod;
+            new_lod.reserve(recursive_sequence_lengths.size());
+            std::copy(recursive_sequence_lengths.begin(),
+                      recursive_sequence_lengths.end(),
+                      std::back_inserter(new_lod));
+            LoD new_offset_lod = ConvertToOffsetBasedLoD(new_lod);
+            PADDLE_ENFORCE_EQ(
+                CheckLoD(new_offset_lod, -1), true,
+                platform::errors::InvalidArgument(
+                    "The provided recursive_sequence_lengths info is invalid"));
```
Comment (chenwhql) [inline paddle/fluid/pybind/pybind.cc]: here can add what the invalid lod is

File: paddle/fluid/pybind/pybind.cc [hunk 6]
```diff
@@ -688,7 +702,8 @@ PYBIND11_MODULE(core_noavx, m) {
              std::copy(lod.begin(), lod.end(), std::back_inserter(new_lod));
              PADDLE_ENFORCE_EQ(
                  CheckLoD(new_lod, vectorize(self.dims()).front()), true,
-                 "the provided lod info is invalid");
+                 platform::errors::InvalidArgument(
+                     "the provided lod info is invalid"));
```
Comment (chenwhql) [inline paddle/fluid/pybind/pybind.cc]: same above

File: paddle/fluid/pybind/pybind.cc [hunk 7]
```diff
@@ -711,23 +726,25 @@ PYBIND11_MODULE(core_noavx, m) {
...
+            LoD new_lod;
+            new_lod.reserve(recursive_sequence_lengths.size());
+            std::copy(recursive_sequence_lengths.begin(),
+                      recursive_sequence_lengths.end(),
+                      std::back_inserter(new_lod));
+            LoD new_offset_lod = ConvertToOffsetBasedLoD(new_lod);
+            PADDLE_ENFORCE_EQ(
+                CheckLoD(new_offset_lod, vectorize(self.dims()).front()), true,
+                platform::errors::InvalidArgument(
+                    "the provided recursive_sequence_lengths info is invalid"));
```
Comment (chenwhql) [inline paddle/fluid/pybind/pybind.cc]: same above

File: paddle/fluid/pybind/pybind.cc [hunk 8]
```diff
@@ -1384,17 +1412,20 @@ All parameter, weight, gradient are variables in Paddle.
...
+                  [](py::bytes protobin) {
+                    proto::OpDesc desc;
+                    PADDLE_ENFORCE_EQ(desc.ParsePartialFromString(protobin),
+                                      true,
+                                      platform::errors::InvalidArgument(
+                                          "Cannot parse user input to OpDesc"));
+                    PADDLE_ENFORCE_EQ(
+                        desc.IsInitialized(), true,
+                        platform::errors::InvalidArgument(
+                            "User OpDesc is not initialized, the reason is: %s",
```
Comment (chenwhql) [inline paddle/fluid/pybind/pybind.cc]: User's OpDesc?

File: paddle/fluid/pybind/tensor_py.h [hunk 9]
```diff
@@ -119,22 +120,29 @@ inline std::string TensorDTypeToPyDTypeStr(
...
+              typeid(T).name()));                                           \
       return py::format_descriptor<T>::format();                            \
     }                                                                       \
   }
 
   _ForEachDataType_(TENSOR_DTYPE_TO_PY_DTYPE);
 #undef TENSOR_DTYPE_TO_PY_DTYPE
-  PADDLE_THROW("Unsupported data type %d", static_cast<int>(type));
+  PADDLE_THROW(platform::errors::Unimplemented(
+      "Unsupported data type %s", framework::DataTypeToString(type)));
```
Comment (chenwhql) [inline paddle/fluid/pybind/tensor_py.h]: `Unsupported data type` -> `Unsupported tensor data type`

File: paddle/fluid/pybind/tensor_py.h [hunk 10]
```diff
@@ -151,7 +159,9 @@ T TensorGetElement(const framework::Tensor &self, size_t offset) {
 
 template <typename T>
 void TensorSetElement(framework::Tensor *self, size_t offset, T elem) {
-  PADDLE_ENFORCE_LT(offset, self->numel());
+  PADDLE_ENFORCE_LT(
+      offset, self->numel(),
+      platform::errors::InvalidArgument("The offset exceed size of tensor."));
```
Comment (chenwhql) [inline paddle/fluid/pybind/tensor_py.h]: `exceed` -> `exceeds`?

File: paddle/fluid/pybind/tensor_py.h [hunk 11]
```diff
@@ -389,7 +402,8 @@ void _sliceDapper(const framework::Tensor *in, framework::Tensor *out,
       _sliceCompute<T, 9>(in, out, ctx, axes, starts);
       break;
     default:
-      PADDLE_THROW("dim size not exepected, current is %d", size);
+      PADDLE_THROW(platform::errors::InvalidArgument(
+          "dim size should be 1 to 9, current is %d", size));
```
Comment (chenwhql) [inline paddle/fluid/pybind/tensor_py.h]: first letter use uppercase?

**Review by chenwhql — COMMENTED** [review]
Maybe we can add a `.` to the end of the error message.

File: paddle/fluid/pybind/tensor_py.h (hunk 10 above)
Comment (zhiqiu) [reply paddle/fluid/pybind/tensor_py.h]: Done, thx.

File: paddle/fluid/pybind/data_set_py.cc (hunk 3 above)
Comment (zhiqiu) [reply paddle/fluid/pybind/data_set_py.cc]: Done, thx.

File: paddle/fluid/pybind/data_set_py.cc (hunk 1 above)
Comment (zhiqiu) [reply paddle/fluid/pybind/data_set_py.cc]: Done, thx.

File: paddle/fluid/pybind/data_set_py.cc (hunk 2 above)
Comment (zhiqiu) [reply paddle/fluid/pybind/data_set_py.cc]: Done, thx.

File: paddle/fluid/pybind/pybind.cc (hunk 4 above)
Comment (zhiqiu) [reply paddle/fluid/pybind/pybind.cc]: Done, thx. The type of an `PyObject* obj `can be get by `obj->ob_type->tp_name`.

File: paddle/fluid/pybind/pybind.cc (hunk 5 above)
Comment (zhiqiu) [reply paddle/fluid/pybind/pybind.cc]: Done, thx.

File: paddle/fluid/pybind/pybind.cc (hunk 6 above)
Comment (zhiqiu) [reply paddle/fluid/pybind/pybind.cc]: Done, thx.

File: paddle/fluid/pybind/pybind.cc (hunk 7 above)
Comment (zhiqiu) [reply paddle/fluid/pybind/pybind.cc]: Done, thx,

File: paddle/fluid/pybind/pybind.cc (hunk 8 above)
Comment (zhiqiu) [reply paddle/fluid/pybind/pybind.cc]: Done, thx.

File: paddle/fluid/pybind/tensor_py.h (hunk 9 above)
Comment (zhiqiu) [reply paddle/fluid/pybind/tensor_py.h]: Done, thx.

File: paddle/fluid/pybind/tensor_py.h (hunk 11 above)
Comment (zhiqiu) [reply paddle/fluid/pybind/tensor_py.h]: Done, thx.

**Review by chenwhql — APPROVED** [review]
LGTM

**Pull request merged**